magento/magento2
# Intoroduced MediaGallerySynchronizationMetadata module

### Description (*)

Intoroduced `MediaGallerySynchronizationMetadata` module eliminating dependency of `MediaGallerySynchronization` on `MediaGalleryMetadataApi` module

Approval Request: https://jira.corp.magento.com/browse/MC-37071

### Related Pull Requests
https://github.com/magento/partners-magento2-infrastructure/pull/29
https://github.com/magento/partners-magento2ee/pull/313


### Fixed Issues (if relevant)

https://jira.corp.magento.com/browse/MC-36993 

### Manual testing scenarios (*)

1. Upgrade Magento 2.4.0 + Adobe Stock Integration 2.0.0 to Magento PR branch + Adobe Stock Integration 2.1.0
2. Verify uploading images with metadata to the new media gallery
3. Verify uploading images with metadata outside of media gallery

### Git based upgrade scenario

```
git clone git@github.com:magento/magento2.git
cd magento2
git checkout 2.4.0
git clone git@github.com:magento/adobe-stock-integration.git ext/magento/stock-integration
cd ext/magento/stock-integration/
git checkout 2.0.0-master
cd ../../..
composer install
composer config minimum-stability dev
composer config repositories.ext path "./ext/*/*/*"
composer require magento/adobe-stock-integration
bin/magento setup:install
git checkout composer.json
git checkout composer.lock
composer install
git remote add sivaschenko git@github.com:sivaschenko/magento2.git
git fetch sivaschenko
git checkout media-gallery-synchronization-metadata
cd ext/magento/stock-integration/
git checkout 2.1.0-master
cd ../../..
git clone git@github.com:magento/adobe-ims.git ext/magento/adobe-ims
composer install
composer config minimum-stability dev
composer config repositories.ext path "./ext/*/*/*"
rm vendor/magento/module-adobe-ims*
composer require magento/adobe-stock-integration
bin/magento module:enable --all
bin/magento setup:upgrade
```

### Contribution checklist (*)
